@prisma/internals doesn't list `ts-toolbelt` in package.json while it is consuming the package

### Bug description

ts-toolbelt is used in the following file in the `@prisma/internals` package: https://github.com/prisma/prisma/blob/445785f6d1d737fc2cb435b1f3b8054c2bf2d6c6/packages/internals/src/cli/checkUnsupportedDataProxy.ts#L3

This package is not listed in the package.json file for @prisma/internals: https://github.com/prisma/prisma/blob/main/packages/internals/package.json. This is causing TS compilation error while consuming this package in a generator. Right now as a workaround, the generator needs to install this dependency. 

### How to reproduce

Try using @prisma/internals package in a typescript starter project. You will receive the following compilation error: 
```
yarn run v1.22.17
$ tsc -d && cpy src/styles dist/styles
node_modules/@prisma/internals/dist/cli/checkUnsupportedDataProxy.d.ts:1:19 - error TS2307: Cannot find module 'ts-toolbelt' or its corresponding type declarations.

1 import { O } from 'ts-toolbelt';
                    ~~~~~~~~~~~~~


Found 1 error in node_modules/@prisma/internals/dist/cli/checkUnsupportedDataProxy.d.ts:1
```

`tsconfig.json` used:
```json
{
  "compilerOptions": {
    "target": "es2019",
    "baseUrl": "src",
    "outDir": "dist",
    "sourceMap": true,
    "module": "commonjs",
    "moduleResolution": "node",
    "lib": ["es2019"], 
    "strict": true,
    "esModuleInterop": true
  }
}
```

### Expected behavior

The above package should list all the dependencies in the `package.json` file.

### Prisma information

Not relevant

### Environment & setup

- OS: MacOS 13.0.1
- Database: Not relevant
- Node.js version: v18.14.0


### Prisma Version

```
prisma                  : 4.10.1
@prisma/client          : 4.10.1
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine aead147aa326ccb985dcfed5b065b4fdabd44b19 (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli aead147aa326ccb985dcfed5b065b4fdabd44b19 (at node_modules/@prisma/engines/migration-engine-darwin-arm64)
Format Wasm             : @prisma/prisma-fmt-wasm 4.10.1-1.80b351cc7c06d352abe81be19b8a89e9c6b7c110
Default Engines Hash    : aead147aa326ccb985dcfed5b065b4fdabd44b19
Studio                  : 0.481.0
```

--- a/packages/client/src/generation/TSClient/TSClient.ts
+++ b/packages/client/src/generation/TSClient/TSClient.ts
@@ -4,7 +4,7 @@ import ciInfo from 'ci-info'
 import crypto from 'crypto'
 import indent from 'indent-string'
 import path from 'path'
-import { O } from 'ts-toolbelt'
+import type { O } from 'ts-toolbelt'
 
 import type { GetPrismaClientConfig } from '../../runtime/getPrismaClient'
 import { DMMFHelper } from '../dmmf'

--- a/packages/client/src/generation/generateClient.ts
+++ b/packages/client/src/generation/generateClient.ts
@@ -25,7 +25,7 @@ import { ensureDir } from 'fs-extra'
 import { bold, dim, green, red } from 'kleur/colors'
 import path from 'path'
 import pkgUp from 'pkg-up'
-import { type O } from 'ts-toolbelt'
+import type { O } from 'ts-toolbelt'
 
 import clientPkg from '../../package.json'
 import type { DMMF as PrismaClientDMMF } from './dmmf-types'

--- a/packages/client/tests/functional/_utils/defineMatrix.ts
+++ b/packages/client/tests/functional/_utils/defineMatrix.ts
@@ -1,4 +1,4 @@
-import { U } from 'ts-toolbelt'
+import type { U } from 'ts-toolbelt'
 
 import { NamedTestSuiteConfig, TestSuiteMatrix } from './getTestSuiteInfo'
 import { setupTestSuiteMatrix, TestCallbackSuiteMeta } from './setupTestSuiteMatrix'

--- a/packages/client/tests/functional/_utils/relationMode/conditionalError.ts
+++ b/packages/client/tests/functional/_utils/relationMode/conditionalError.ts
@@ -1,4 +1,4 @@
-import { O } from 'ts-toolbelt'
+import type { O } from 'ts-toolbelt'
 
 import { AdapterProviders, Providers } from '../providers'
 

--- a/packages/internals/package.json
+++ b/packages/internals/package.json
@@ -75,6 +75,7 @@
     "tmp": "0.2.3",
     "ts-node": "10.9.2",
     "ts-pattern": "5.2.0",
+    "ts-toolbelt": "9.6.0",
     "typescript": "5.4.5",
     "yarn": "1.22.22"
   },

--- a/packages/internals/src/cli/checkUnsupportedDataProxy.ts
+++ b/packages/internals/src/cli/checkUnsupportedDataProxy.ts
@@ -1,6 +1,6 @@
 import fs from 'fs'
 import { green } from 'kleur/colors'
-import { O } from 'ts-toolbelt'
+import type { O } from 'ts-toolbelt'
 
 import { getConfig, getEffectiveUrl, getSchemaWithPath, link } from '..'
 import { resolveUrl } from '../engine-commands/getConfig'

